### ISSUE TITLE
Latest Version

### ISSUE THREAD

**Issue by smolck** [gilligan/nix-neovim-nightly#2]
@gilligan The current latest version of neovim-nightly is 0.5.0. Would you mind updating this repo accordingly? I'd really like to use the latest neovim version but am new to Nix, so an update here would help out a bunch!

**Comment by gilligan** [comment]
Wanted to update it just now but realized something is breaking and i'd have to take a closer look. I'll try to do so when i find time. Just can't do it right away now - sorry.

**Comment by smolck** [comment]
No problem! Actually, @teto from the Neovim Gitter helped me out tremendously to get things working with a simple overlay (and *without* this repo). One just needs to put the following (or something similar) in `~/.config/nixpkgs/overlays/neovim.nix` and then install `neovim` as normal (for ex. with `nix-env -iA nixos.neovim` on NixOS; also, probably needs to be a global overlay if installing system-wide):
```nix
# ~/.config/nixpkgs/overlays/neovim.nix

self: prev: {
  neovim-unwrapped = (prev.neovim-unwrapped.override { lua = self.luajit; }).overrideAttrs(oldAttrs: {
    cmakeBuildType="debug";
    cmakeFlags = oldAttrs.cmakeFlags ++ [ "-DMIN_LOG_LEVEL=0" ];

    version = "master";
    src = builtins.fetchGit {
      url = https://github.com/neovim/neovim.git;
    };

    nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [
      # self.pkgs.valgrind

      # testing between both
      self.pkgs.ccls
      self.pkgs.clang-tools  # for clangd
    ];

    buildInputs = oldAttrs.buildInputs ++ [
      self.pkgs.icu  # for treesitter unicode/ptypes.h
      self.pkgs.utf8proc
    ];

    # export NVIM_PROG
    # https://github.com/neovim/neovim/blob/master/test/README.md#configuration
    shellHook = oldAttrs.shellHook + ''
      export NVIM_PYTHON_LOG_LEVEL=DEBUG
      export NVIM_LOG_FILE=/tmp/log
      # export NVIM_PYTHON_LOG_FILE=/tmp/log
      export VALGRIND_LOG="$PWD/valgrind.log"
      export NVIM_TEST_PRINT_I=1
      export NVIM_TEST_MAIN_CDEFS=1
      echo "To run tests:"
      echo "VALGRIND=1 TEST_FILE=test/functional/core/job_spec.lua TEST_TAG=env make functionaltest"
    '';
  });
}
```

Note again, that this is not my work, but @teto's. I may also have gotten something wrong or not quite right above, but it works for me. This may or may not be helpful to you or this project, but figured I'd post it since it really helped me out.

**Comment by sondr3** [comment]
Isn't that building a debug version of neovim and not a release version? I've been looking for a few hours now for a reasonably up-to-date `master` version of neovim but haven't been able to find anything recent, I saw your fork of this repo @smolck but it installed `0.4.0-dev` for some reason.

**Comment by smolck** [comment]
@sondr3 I *presume* it’s doing a release build, since it’s only an overlay for the `neovim-unwrapped` package defined here (which I would imagine is doing a release build): https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/neovim/default.nix 
You can check that file I linked to verify for sure that it is a release build of course.

BTW, don’t mind my fork. I didn’t (and still don’t really) understand what’s going on in this repo, so it was more of a testing ground for myself.

**Comment by sondr3** [comment]
@smolck alright, cool. I'm still unable to build it though using your overlay posted here, I keep getting `Could NOT find LibLUV (missing: LIBLUV_LIBRARY LIBLUV_INCLUDE_DIR)`, are you on `stable` or `unstable`?

**Comment by smolck** [comment]
@sondr3 I'm on `unstable`. It appears btw that yes, the overlay builds a debug build as you thought. However, I think that could be changed by slightly modifying the overlay, although I'm not sure exactly how atm.

**Comment by smolck** [comment]
@sondr3 After asking @teto on Gitter (it's his overlay, not mine), he said to remove `cmakeBuildType="debug";` from the overlay and the build should be a release build.

**Comment by sondr3** [comment]
@smolck thanks, I managed to get it working by switching to `unstable` as my default channel (which I've been meaning to do anyways). Thanks a bunch for your help!

**Comment by teto** [comment]
@sondr3 you might not know this but you can install neovim from unstable while keeping the rest of your system on stable: see https://github.com/teto/home/blob/master/nixpkgs/home-desktop.nix#L4 for an example.

**Comment by sondr3** [comment]
@teto yeah, I've been doing it with a few packages that have been broken/missing on `stable`. But as mentioned I've been meaning to move to `unstable` regardless as I want things to be a bit more up to date.

**Comment by NickHu** [comment]
FYI in case anyone else is looking to try neovim-0.5, the following simple overlay works now:

```nix
self: super: {
  neovim-unwrapped = super.neovim-unwrapped.overrideAttrs (oldAttrs: {
    version = "master";
    src = builtins.fetchGit {
      url = https://github.com/neovim/neovim.git;
    };
  });
}
```

Seems like all the build problems that caused this repo, and also the move to LuaJIT, have been resolved in nixpkgs

**Comment by gilligan** [comment]
I definitely need to update this repo or just drop it 😅

**Comment by mjlbach** [comment]
Maybe we could move this into the nix-community build cache?

**Comment by NickHu** [comment]
@mjlbach As per my previous comment, all of the patches in this repo are obsolete
